Weather-Monitoring-And-Reporting-Service_Tests/TextFormatStrategyFactoryTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿using FluentAssertions;
+
+using Weather_Monitoring_And_Reporting_Service.Strategies;
+using Weather_Monitoring_And_Reporting_Service;
+
+namespace Weather_Monitoring_And_Reporting_Service_Tests;
+
+public class TextFormatStrategyFactoryTests
+{
+    [Theory]
+    [InlineData(null)]
+    [InlineData("")]
+    public void GetTextFormatStrategy_EmptyOrNullString_ThrowsException(string? invalidWeatherDataFilePath)
+    {
+        // Arrange & Act
+        Action act = () => TextFormatStrategyFactory.GetTextFormatStrategy(invalidWeatherDataFilePath);
+
+        // Assert
+        act.Should().Throw<WeatherDataFilePathNullException>()
+            .WithMessage("Weather data file path is null or empty.");
```

Comment:
assume we have changed the message, this test will fail and we will need to fix it :) . Find a solution for this